src-plugins/itkN4BiasCorrection/itkN4BiasCorrection.cpp

```diff
@@ -86,68 +84,64 @@ void itkN4BiasCorrection::setParameter ( double  data, int channel )
 
 int itkN4BiasCorrection::update()
 {
-    if ( !d->input )
-        return -1;
-
-    QString id = d->input->identifier();
+    int res = DTK_FAILURE;
 
-    qDebug() << "itkFilters, update : " << id;
-
-    if ( id == "itkDataImageChar3" )
-    {
-        d->update<char>();
-    }
-    else if ( id == "itkDataImageUChar3" )
-    {
-        d->update<unsigned char>();
-    }
-    else if ( id == "itkDataImageShort3" )
-    {
-        d->update<short>();
-    }
-    else if ( id == "itkDataImageUShort3" )
-    {
-        d->update<unsigned short>();
-    }
-    else if ( id == "itkDataImageInt3" )
-    {
-        d->update<int>();
-    }
-    else if ( id == "itkDataImageUInt3" )
-    {
-        d->update<unsigned int>();
-    }
-    else if ( id == "itkDataImageLong3" )
-    {
-        d->update<long>();
-    }
-    else if ( id== "itkDataImageULong3" )
-    {
-        d->update<unsigned long>();
-    }
-    else if ( id == "itkDataImageFloat3" )
-    {
-        d->update<float>();
-    }
-    else if ( id == "itkDataImageDouble3" )
-    {
-        d->update<double>();
-    }
-    else
+    if (d->input)
     {
-        qDebug() << description()
-                 <<", Error : pixel type not yet implemented ("
-                 << id
-                 << ")";
-        return -1;
+        QString id = d->input->identifier();
+
+        if ( id == "itkDataImageChar3" )
+        {
+            res = d->update<char>();
+        }
+        else if ( id == "itkDataImageUChar3" )
+        {
+            res = d->update<unsigned char>();
+        }
+        else if ( id == "itkDataImageShort3" )
+        {
+            res = d->update<short>();
+        }
+        else if ( id == "itkDataImageUShort3" )
+        {
+            res = d->update<unsigned short>();
+        }
+        else if ( id == "itkDataImageInt3" )
+        {
+            res = d->update<int>();
+        }
+        else if ( id == "itkDataImageUInt3" )
+        {
+            res = d->update<unsigned int>();
+        }
+        else if ( id == "itkDataImageLong3" )
+        {
+            res = d->update<long>();
+        }
+        else if ( id== "itkDataImageULong3" )
+        {
+            res = d->update<unsigned long>();
+        }
+        else if ( id == "itkDataImageFloat3" )
+        {
+            res = d->update<float>();
+        }
+        else if ( id == "itkDataImageDouble3" )
+        {
+            res = d->update<double>();
+        }
+        else
+        {
+            res = medAbstractProcess::PIXEL_TYPE;
```

Comment:
You are mixing two different return types: success state (DTK_SUCCESS or DTK_FAILURE) and error type (medAbstractProcess::DataError enum).